lib/libutils/ext/trace.c

```diff
@@ -92,34 +91,31 @@ void trace_printf(const char *function, int line, int level, bool level_ok,
 		thread_id = trace_ext_get_thread_id();
 
 	if (thread_id >= 0) {
-		res = snprintk(buf + boffs, sizeof(buf) - boffs, "[0x%x] ",
+		res = snprintk(buf + boffs, sizeof(buf) - boffs, "%d",
```

Comment:
So, this is the change going from my initial proposal:
`<type>/<where> [<thread_id>] [<func:line>] <message>`
to
`<type><thread_id><where> [<func:line>] <message>`

I don't really like baking the thread_id into the print. Even though it save a few characters, it makes the OP-TEE debug messages too dynamic. I.e., we get something like this
```bash
D0TC gic_it_set_cpu_mask:270 cpu_mask: 0x0
D0TC gic_it_set_prio:283 prio: writing 0x1 to 0x11d00428
D0TC context_init:101 0 reader detected
ITC Initialized
D0TC init_primary_helper:853 Primary CPU switching to normal world boot
ITC Dynamic shared memory is enabled
D0TC core_mmu_alloc_l2:249 L2 table used: 4/4
D0TC tee_ta_init_pseudo_ta_session:292 Lookup pseudo TA 484d4143-2d53-4841-3120-4a6f636b6542
D0TC tee_ta_init_user_ta_session:636 Lookup user TA 484d4143-2d53-4841-3120-4a6f636b6542 (REE)
D0TC ta_load:316 ELF load address 0x101000
DTA register_shared_key:147 Got shared key 12345678901234567890 (20 bytes).
E0TC 
E0TC TA panicked with code 0x0
E0TC Status of TA 484d4143-2d53-4841-3120-4a6f636b6542 (0xe07ba50) (active)
E0TC  arch: arm  load address: 0x101000  ctx-idr: 1
E0TC  stack: 0x100000 4096
E0TC  region 0: va 0x100000 pa 0xe31d000 size 0x1000 flags rw-
E0TC  region 1: va 0x101000 pa 0xe300000 size 0xf000 flags r-x
E0TC  region 2: va 0x110000 pa 0xe30f000 size 0x3000 flags r--
E0TC  region 3: va 0x113000 pa 0xe312000 size 0xb000 flags rw-
E0TC  region 4: va 0 pa 0 size 0 flags ---
E0TC  region 5: va 0 pa 0 size 0 flags ---
E0TC  region 6: va 0 pa 0 size 0 flags ---
E0TC  region 7: va 0 pa 0 size 0 flags ---
E0TC Call stack:
E0TC  0x001044a8
E0TC  0x0010ba0d
E0TC  0x00101397
E0TC  0x00101545
E0TC  0x0010441b
E0TC  0x00104477
D0TC user_ta_enter:452 tee_user_ta_enter: TA panicked with code 0x0
D0TC tee_ta_invoke_command:649 Error: ffff3024 of 3
D0TC tee_ta_close_session:402 tee_ta_close_session(0xe07be98)
D0TC tee_ta_close_session:421 Destroy session
D0TC tee_ta_close_session:447 Destroy TA ctx
```
compared to this in the initial patch

```bash
D/TC [00] gic_it_set_cpu_mask:270 cpu_mask: 0x0
D/TC [00] gic_it_set_prio:283 prio: writing 0x1 to 0x11d00428
D/TC [00] context_init:101 0 reader detected
I/TC Initialized
D/TC [00] init_primary_helper:853 Primary CPU switching to normal world boot
I/TC Dynamic shared memory is enabled
D/TC [00] core_mmu_alloc_l2:249 L2 table used: 4/4
D/TC [00] tee_ta_init_pseudo_ta_session:292 Lookup pseudo TA 484d4143-2d53-4841-3120-4a6f636b6542
D/TC [00] tee_ta_init_user_ta_session:636 Lookup user TA 484d4143-2d53-4841-3120-4a6f636b6542 (REE)
D/TC [00] ta_load:316 ELF load address 0x101000
D/TA register_shared_key:147 Got shared key 12345678901234567890 (20 bytes).
E/TC [00] 
E/TC [00] TA panicked with code 0x0
E/TC [00] Status of TA 484d4143-2d53-4841-3120-4a6f636b6542 (0xe07ba50) (active)
E/TC [00]  arch: arm  load address: 0x101000  ctx-idr: 1
E/TC [00]  stack: 0x100000 4096
E/TC [00]  region 0: va 0x100000 pa 0xe31d000 size 0x1000 flags rw-
E/TC [00]  region 1: va 0x101000 pa 0xe300000 size 0xf000 flags r-x
E/TC [00]  region 2: va 0x110000 pa 0xe30f000 size 0x3000 flags r--
E/TC [00]  region 3: va 0x113000 pa 0xe312000 size 0xb000 flags rw-
E/TC [00]  region 4: va 0 pa 0 size 0 flags ---
E/TC [00]  region 5: va 0 pa 0 size 0 flags ---
E/TC [00]  region 6: va 0 pa 0 size 0 flags ---
E/TC [00]  region 7: va 0 pa 0 size 0 flags ---
E/TC [00] Call stack:
E/TC [00]  0x001044a8
E/TC [00]  0x0010ba11
E/TC [00]  0x00101397
E/TC [00]  0x00101545
E/TC [00]  0x0010441b
E/TC [00]  0x00104477
D/TC [00] user_ta_enter:452 tee_user_ta_enter: TA panicked with code 0x0
D/TC [00] tee_ta_invoke_command:649 Error: ffff3024 of 3
D/TC [00] tee_ta_close_session:402 tee_ta_close_session(0xe07be98)
D/TC [00] tee_ta_close_session:421 Destroy session
D/TC [00] tee_ta_close_session:447 Destroy TA ctx
```
I would like to go with the latter, since I think that make OP-TEE debug messages more distinct (and it also make regexp grep less complicated whenever that is needed, symbolize.py, HAB-test etc). I.e, I won't to remove/revert the `[review]: Move thread ID` patch again.

Reply:
My proposal was to print a `/` when thread ID wasn't available to keep the positions constant with or without thread id available. Also only one digit is needed for the thread ID.
I think it's pretty useless to be able to not print the thread ID if it's available as that's important debug information.
If I'd have to chose between knowing loglevel + `trace_ext_prefix` versus thread ID the choice would be easy.

Reply:
I think the thread id is easier to catch from the 2nd format: `E/TC [<thread-id>] ...`

Reply:
So, what about something in-between our proposals?
`E/TC:0`
or
`E/TC 0`

That reduces it quite a bit but still keeps the original patch "prefix". This way the regexp in symbolize doesn't have to care about the number of characters for thread_id's and the same when searching for patterns from our test environment.

Reply:
Sure, but I don't see the problem with updating the regexp.
By the way, what do you propose to print when thread ID isn't available?

Reply:
When thread ID isn't available I propose to print nothing as done in the output above from the initial patch.

Reply:
That means left shifting those traces. Depending on what's traced that could be annoying.

Reply:
Ok, then I'll print a single `" "` there instead and we are good to go.

Reply:
So, can we decide whether we shall a single or two digits? My initial proposal was two, but I've hearing back and forth. Shall we say two digits again then?

Reply:
In that case it'd look like this
```bash
D/TC:00 init_canaries:171 watch *0xe085dfc
D/TC:00 init_canaries:173 #Stack canaries for stack_thread[0] with top at 0xe087e38
D/TC:00 init_canaries:173 watch *0xe087e3c
D/TC:00 init_canaries:173 #Stack canaries for stack_thread[1] with top at 0xe089e78
D/TC:00 init_canaries:173 watch *0xe089e7c
I/TC:   OP-TEE version: 2.6.0-99-gea475ccd-dev #5 Fri Dec 15 10:53:26 UTC 2017 arm
D/TC:00 tee_ta_register_ta_store:608 Registering TA store: 'REE' (priority 10)
D/TC:00 mobj_mapped_shm_init:579 Shared memory address range: e300000, 10300000
D/TC:00 gic_it_set_cpu_mask:266 cpu_mask: writing 0xff to 0x11d00828
D/TC:00 gic_it_set_cpu_mask:270 cpu_mask: 0x0
D/TC:00 gic_it_set_prio:283 prio: writing 0x1 to 0x11d00428
D/TC:00 context_init:101 0 reader detected
I/TC:   Initialized
D/TC:00 init_primary_helper:853 Primary CPU switching to normal world boot
I/TC:   Dynamic shared memory is enabled
```

Reply:
The number of cores is not directly connected to the number of threads.
Since this is a hard coded number we could use one or two digits depending on the number of configured threads.

Reply:
Ok, everyone happy with that (I think it's overkill, but ...)? I'll add some ifdefs in the code and sort it out.